Allow bagging to destination

nkrabben opened the pull request:
Based on @runderwood #92, but reimplemented within the current codebase to address #32 and #136 

Major difference, this places every source within its own bag rather than combining bag, e.g.
`bagit.py --destination foo/ --directory bar/ baz/`
will create 
```
foo
├──bar
│  ├── bag-info.txt
│  ├── ...
└──baz
   ├── bag-info.txt
   ├── ...
```
which mirrors current behavior that bagit.py --directory bar/ baz/ creates
```
./
├──bar
│  ├── bag-info.txt
│  ├── ...
└──baz
   ├── bag-info.txt
   ├── ...
```

kba commented on bagit.py:
```diff
@@ -162,30 +163,41 @@ def make_bag(
+        if not os.path.isdir(dest_dir):
+            os.makedirs(dest_dir)
+        else:
+            raise RuntimeError(_("The following directory already exists:\n%s"), dest_dir)
```
Consider allowing an empty directory as a target, useful e.g. when creating that directory requires different permissions than the user running bagit has.

nkrabben replied:
Good point. Would checking for an empty directory like this be enough?
```suggestion
      elif len(os.listdir(dest_dir)) > 0:
            raise RuntimeError(_("The following directory already exists and contains files:\n%s"), dest_dir)
```

acdha replied:
That will work because `os.listdir` is documented as not including `.` or `..` but I was wondering about performance if someone points it at a directory which a large number of files — the classic Unix answer being `os.stat(dest_dir).st_nlink > 2` — but I don't think that'll really be an issue in this scenario.

kba commented on bagit.py:
```diff
@@ -162,30 +163,41 @@ def make_bag(
+    else:
+        dest_dir = os.path.abspath(bag_dir)
+
+    source_dir = os.path.abspath(bag_dir)
```
Consider using `realpath` instead to resolve symlinks (not related to this change, just a general observation)

nkrabben replied:
I couldn't think of any edge cases, but I did have to invoke realpath in the setup class for the tests.

nkrabben commented:
I've incorporated the suggested changes.

After using this branch for a few months, I changed the copy mechanism to use a single copytree call. To do this, I had to create the temporary directory name from scratch since copytree will not copy to pre-existing directories. I couldn't think of a way to cover L245-248 with a test.

nkrabben commented:
@acdha is it okay if I add tests to cover the CLI? It would have caught the bug I fixed in 61769cd, but I hadn't seen an approach to testing in the package yet.

acdha commented:
> @acdha is it okay if I add tests to cover the CLI? It would have caught the bug I fixed in [61769cd](https://github.com/LibraryOfCongress/bagit-python/commit/61769cda626c8493b6a733885a407915f4ada63f), but I hadn't seen an approach to testing in the package yet.

Definitely – that's a big gap in the test coverage currently
